### PR TITLE
tests: perform graceful rolling restarts in storcon scale test

### PR DESCRIPTION
**Pull request by VladLazar**
## Problem
Scale test doesn't exercise drain & fill.

## Summary of changes
Make scale test exercise drain & fill

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: test_runner/performance/test_storage_controller_scale.py [hunk 1]
```diff
@@ -59,6 +130,9 @@ def test_storage_controller_many_tenants(
         ]
     )
 
+    # TODO: default tenant does not have a secondary location
+    env.storage_controller.allowed_errors.append(".*Scheduling error when draining pageserver.*")
```
Comment (jcsp) [inline test_runner/performance/test_storage_controller_scale.py:134]: I can't think of a good reason that I left the default tenant in this test: we should probably just avoid creating it

File: test_runner/performance/test_storage_controller_scale.py [hunk 2]
```diff
@@ -195,10 +271,44 @@ def check_memory():
...
-        pageserver.start()
+    shard_counts = get_consistent_node_shard_counts(env, total_shards)
+    log.info(f"Shard counts before rolling restart: {shard_counts}")
+
+    assert_consistent_balanced_attachments(env, total_shards)
+
+    # Restart pageservers gracefully: this exercises the /re-attach pageserver API
+    # and the storage controller drain and fill API
+    for ps in env.pageservers:
+        env.storage_controller.retryable_node_operation(
```
Comment (jcsp) [inline test_runner/performance/test_storage_controller_scale.py:282]: How come this needs the retryable_node_operation wrapper?  Can we make it more specific for which kinds of error it's meant to retry?

**Review by jcsp — APPROVED** [review]

File: test_runner/performance/test_storage_controller_scale.py (hunk 2 above)
Comment (VladLazar) [reply test_runner/performance/test_storage_controller_scale.py:282]: It doesn't need it in this controlled environment. I followed the RFC specification for how deployment orchestrator should do.

File: test_runner/performance/test_storage_controller_scale.py (hunk 1 above)
Comment (VladLazar) [reply test_runner/performance/test_storage_controller_scale.py]: [622e255](https://github.com/neondatabase/neon/pull/8173/commits/622e255ffe4d6748d09cec0b0c83381daf1afc4d)

**Comment by VladLazar** [comment]
Don't get why https://github.com/neondatabase/neon/actions/runs/9777402273/job/26996882519 failed when the report was generated. Merging regardless.

**Pull request merged**